pkg/fleetautoscalers/fleetautoscalers_test.go

```diff
@@ -152,55 +79,592 @@ func (t testServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+			webhookPolicy: &autoscalingv1.WebhookPolicy{
+				Service: nil,
+				URL:     &invalidURL,
+			},
+			expected: expected{
+				replicas: 0,
+				limited:  false,
+				err:      "parse \")1golang.org/\": invalid URI for request",
+			},
+		},
+		{
+			description: "Service name is empty",
+			webhookPolicy: &autoscalingv1.WebhookPolicy{
+				Service: &admregv1b.ServiceReference{
+					Name:      "",
+					Namespace: "default",
+					Path:      &url,
+				},
+			},
+			expected: expected{
+				replicas: 0,
+				limited:  false,
+				err:      "service name was not provided",
+			},
+		},
+		{
+			description: "No certs",
+			webhookPolicy: &autoscalingv1.WebhookPolicy{
+				Service: &admregv1b.ServiceReference{
+					Name:      "service1",
+					Namespace: "default",
+					Path:      &url,
+				},
+				CABundle: []byte("invalid-value"),
+			},
+			expected: expected{
+				replicas: 0,
+				limited:  false,
+				err:      "no certs were appended from caBundle",
+			},
+		},
+		{
+			description: "Wrong server URL",
+			webhookPolicy: &autoscalingv1.WebhookPolicy{
+				Service: nil,
+				URL:     &wrongServerURL,
+			},
+			expected: expected{
+				replicas: 0,
+				limited:  false,
+				err:      "Post \"http://127.0.0.1:1\": dial tcp 127.0.0.1:1: connect: connection refused",
+			},
+		},
+		{
+			description: "Handle server error",
+			webhookPolicy: &autoscalingv1.WebhookPolicy{
+				Service: nil,
+				URL:     &(server.URL),
+			},
+			specReplicas:   50,
+			statusReplicas: 50,
+			// hardcoded value in a server implementation
+			statusAllocatedReplicas: -10,
+			statusReadyReplicas:     40,
+			expected: expected{
+				replicas: 50,
+				limited:  false,
+				err:      fmt.Sprintf("bad status code %d from the server: %s", http.StatusInternalServerError, server.URL),
+			},
+		},
+		{
+			description: "Handle invalid response from the server",
+			webhookPolicy: &autoscalingv1.WebhookPolicy{
+				Service: nil,
+				URL:     &(server.URL),
+			},
+			specReplicas:            50,
+			statusReplicas:          50,
+			statusAllocatedReplicas: -20,
+			statusReadyReplicas:     40,
+			expected: expected{
+				replicas: 50,
+				limited:  false,
+				err:      "invalid character 'i' looking for beginning of value",
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			f.Spec.Replicas = tc.specReplicas
+			f.Status.Replicas = tc.statusReplicas
+			f.Status.AllocatedReplicas = tc.statusAllocatedReplicas
+			f.Status.ReadyReplicas = tc.statusReadyReplicas
+
+			replicas, limited, err := applyWebhookPolicy(tc.webhookPolicy, f)
+
+			if tc.expected.err != "" && assert.NotNil(t, err) {
+				assert.Equal(t, tc.expected.err, err.Error())
+			} else {
+				assert.Equal(t, tc.expected.replicas, replicas)
+				assert.Equal(t, tc.expected.limited, limited)
+				assert.Nil(t, err)
+			}
+		})
+	}
+}
+
+func TestApplyWebhookPolicyNilFleet(t *testing.T) {
+	t.Parallel()
+
+	url := "/scale"
```

Comment:
```suggestion
	url := "scale"
```
Can you please check if we can omit redundant slash sign.

Reply:
Yes, we can, it is mentioned in docs. I will remove this slash.